#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Un-hide errno
# DP: Author: Daniel Jacobowitz <dan@debian.org>
# DP: Upstream status: Not submitted
# DP: Status Details: This patch makes errno re-available at link time.
# DP:    The right solution is to make it only available at run time without
# DP:    re-breaking old applications.
# DP:    This patch should not be considered permanent; see libc-alpa
# DP:    archives for 12/23/2002 for a better alternative (not implemented
# DP:    yet)
# DP: Date: Dec 24, 2002

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- glibc-2.3.1/sysdeps/generic/errno.c.orig	2002-12-24 10:08:44.000000000 -0500
+++ glibc-2.3.1/sysdeps/generic/errno.c	2002-12-24 10:09:01.000000000 -0500
@@ -36,7 +36,7 @@
    visible at link time.  Programs must use the accessor functions.  */
 # if defined HAVE_ELF && defined SHARED && defined DO_VERSIONING
 #  include <shlib-compat.h>
-compat_symbol (libc, errno, errno, GLIBC_2_0);
-compat_symbol (libc, _errno, _errno, GLIBC_2_0);
+//compat_symbol (libc, errno, errno, GLIBC_2_0);
+//compat_symbol (libc, _errno, _errno, GLIBC_2_0);
 # endif
 #endif
